### PR TITLE
Update documentation to `cmake` build infrastructure.

### PR DESCRIPTION
**Pull request by tammoippen**
This PR is depending on #213.

It updates the [developer space](https://nest.github.io/nest-simulator/) documentation with respect to building NEST with `cmake`.

### CODE REVIEW

File: extension_modules.md [hunk 1]
```diff
@@ -58,57 +53,44 @@ and other users.
...
-          -> Running aclocal ...
-          -> Installing libtool components ...
-          -> Running autoheader ...
-          -> Running automake ...
-          -> Running autoconf ...
-        Done.
-
-    You may see some more output.
-3. Leave MyModule and create a build directory for it, e.g., mmb next to it
+2. Create a build directory for it on the same level as MyModule, e.g., mmb next to it
```
Comment (jougs) [inline extension_modules.md]: 2. Create a build directory for it on the same level as MyModule (e.g. `mmb`)

File: extension_modules.md [hunk 2]
```diff
@@ -58,57 +53,44 @@ and other users.
...
 
-4. You should now see pif_psc_alpha in the modeldict and drop_odd_spike in the synapsedict. You can learn more about these models and the additional (meaningless) connection function supplied by the model by typing
+3. You should now see `pif_psc_alpha` in the `modeldict` and `drop_odd_spike` in the `synapsedict`. You can learn more about these models and the additional (meaningless) connection function supplied by the model by typing
 
         /pif_psc_alpha help
         /drop_odd_spike help
         /StepPatternConnect help
 
-5. In PyNest, use
+4. In PyNest, use
```
Comment (jougs) [inline extension_modules.md]: PyNEST

File: extension_modules.md [hunk 3]
```diff
@@ -58,57 +53,44 @@ and other users.
 As a starting point, try to build MyModule as follows:
 
 1. From the NEST source directory, copy directory examples/MyModule to somewhere outside the NEST source, build or install directories.
```
Comment (jougs) [inline extension_modules.md]: Please add backticks: `examples/MyModule`

**Comment by jougs** [comment]
:+1: once my comments are addressed. Thanks!

**Comment by tammoippen** [comment]
@jougs done with 	a1f9b38.

**Comment by heplesser** [comment]
:+1: and merging.

**Pull request merged**